lnrpc/routerrpc/router_server.go

```diff
@@ -356,20 +360,62 @@ func (s *Server) SendPaymentV2(req *SendPaymentRequest,
 	)
 }
 
-// EstimateRouteFee allows callers to obtain a lower bound w.r.t how much it
-// may cost to send an HTLC to the target end destination.
+// EstimateRouteFee allows callers to obtain an upper bound w.r.t how much it
+// may cost to send an HTLC to the target end destination. This method sends
+// probe payments to the target node, based on target invoice parameters and a
+// random payment hash that makes it impossible for the target to settle the
+// htlc. If provided with a destination key and amount, this method will perform
+// a local graph based fee estimation.
```

Comment:
this sounds like it should be documented clearly in the proto files. it would surprise me a lot! on a related note, as a caller I would expect to be able to choose whether I want probing or query routes when providing an invoice. for example for time-sensitive contexts where waiting 10s of seconds for a probe is not acceptable.

Reply:
Done.